### PR TITLE
Fixes for on-premises

### LINKED ISSUE #10: Python3 compatibility
When I import this package with Python3, I get a syntax error because ThreeScalePY.py uses the older 'except' syntax, e.g. line 206. I don't think you ever supported Python 2.5, so it should be safe to recode these lines with the newer syntax.

### LINKED ISSUE #24: service_id required
From November 2016 `service_id` is mandatory. The code would need to be reviewed to support the use of service_id (it should be optional to avoid breaking change).

### LINKED ISSUE #25: Allow backend host and Port configurable
The default_host = "su1.3scale.net".
To work with on-prem version, the host and the port would need to be configurable.

### LINKED ISSUE #26: service_token should be supported along with provider_key
The plugin would need to be updated to support service_token along with provider_key.

### PR DESCRIPTION
**Pull request by mayorova**
The fixes included in the PR:
- `service_id` is supported
- support both `provider_key` and `service_token`
-  support custom backend via `backend_url` argument in constructor

### Code:
~- [ ] add a deprecation note for a provider key~
- [x] validate backend_uri in ThreeScale constructor, and throw an error if it's not a valid URI
- [x] bump the release

### Tests:
- [x] a test for ThreeScaleAuthorize.authorize - it should handle 3scale's 403 and 404 error code
- [x] client constructor must throw an error when no provider key and no service_id/service_token provided
- [x] client works with either provider_key or service_token (authorize / report / authrep)
- [x] service_id can be added for both provider_key and service_token

### Documentation:
- [x] update instructions for service_token and service_id
- [x] update instructions for custom host/port
- [x] add a note that `service_id` is required
~- [ ] add a note that `provider_key` is deprecated~
- [x] update release notes

### Release notes:
- [x] authorize() method supports `usage` and `other_params` arguments
- [x] service_token is supported

Closes #24, closes #25, closes #26
Also closes #10

### CODE REVIEW

File: ThreeScalePY.py [hunk 1]
```diff
@@ -84,31 +90,56 @@
           ]
 
 class ThreeScale:
+
+    DEFAULT_BACKEND_URI = 'https://su1.3scale.net:443'
```
Comment (andrewdavidmackenzie) [inline ThreeScalePY.py]: I'd use "Service Management API" for this url/name....not "backend"

Comment (mayorova) [inline ThreeScalePY.py]: Agree, I tried "Service Management API" everywhere in the documentation, but it's a bit too long for the variable name :) Also, won't the component called "backend" in on-premise version? I'm afraid calling it `service_mgmg_api_uri` might be more confusing...

File: README.md [hunk 2]
```diff
@@ -1,67 +1,179 @@
 # Client for 3scale web service management system API [![Build Status](https://secure.travis-ci.org/3scale/3scale_ws_api_for_python.png?branch=master)](http://travis-ci.org/3scale/3scale_ws_api_for_python)
 3scale integration plugin for Python applications. 3scale is an API Infrastructure service which handles API Keys, Rate Limiting, Analytics, Billing Payments and Developer Management. Includes a configurable API dashboard and developer portal CMS. More product stuff at http://www.3scale.net/, support information at http://support.3scale.net/.
 
+# Synopsis
+This plugin supports the 3 main calls to the 3scale backend:
```
Comment (andrewdavidmackenzie) [inline README.md]: At least here I'd name the Service Management API?

File: README.md [hunk 3]
```diff
@@ -1,67 +1,179 @@
...
+Check the [3scale APIs ActiveDocs](https://support.3scale.net/docs/3scale-apis-activedocs) to learn more about the supported parameters.
+
 # Installation:
 
-Standard distutils installation: unpack ThreeScale-2.0.tar.gz, and from ThreeScale-2.0 directory run
+## Using Pip:  
+```shell
+pip install ThreeScalePY
+```
+## Standard distutils installation:
```
Comment (andrewdavidmackenzie) [inline README.md]: Use `setuptools` instead of `distutils` ???

**Review by andrewdavidmackenzie — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (mayorova) [inline README.md]: Yep, thanks!

File: README.md (hunk 2 above)
Comment (mayorova) [inline README.md]: 👍

**Pull request merged**